runtime/src/accounts_db.rs

```diff
@@ -121,31 +122,31 @@ impl Versioned for (u64, AccountInfo) {
 }
 
 #[derive(Clone, Default, Debug)]
-pub struct AccountStorage(pub HashMap<Slot, SlotStores>);
+pub struct AccountStorage(pub DashMap<Slot, Arc<RwLock<SlotStores>>>);
 
 impl AccountStorage {
-    fn scan_accounts(&self, account_info: &AccountInfo, slot: Slot) -> Option<(Account, Slot)> {
+    fn get_account_storage_entry(
+        &self,
+        slot: Slot,
+        store_id: AppendVecId,
+    ) -> Option<Arc<AccountStorageEntry>> {
         self.0
             .get(&slot)
-            .and_then(|storage_map| storage_map.get(&account_info.store_id))
-            .and_then(|store| {
-                Some(
-                    store
-                        .accounts
-                        .get_account(account_info.offset)?
-                        .0
-                        .clone_account(),
-                )
-            })
-            .map(|account| (account, slot))
+            .and_then(|storage_map| storage_map.value().read().unwrap().get(&store_id).cloned())
+    }
+
+    fn get_slot_stores(&self, slot: Slot) -> Option<Arc<RwLock<SlotStores>>> {
```

Comment:
nits: how about adjusting the type alias `SlotStores` and hide `Arc<RwLock<_>>` here?